pkg/operator/contenthash/topology_spread_constraints.go

```diff
@@ -42,7 +43,16 @@ func TopologySpreadConstraints(in []corev1.TopologySpreadConstraint) string {
 			labelSelectors = map[string]string{}
 		}
 		writeStringHash(h, StringMap(labelSelectors))
-
+		writeStringHash(h, strings.Join(tsc.MatchLabelKeys, ""))
+		if tsc.MinDomains != nil {
+			writeStringHash(h, string(*tsc.MinDomains))
+		}
+		if tsc.NodeAffinityPolicy != nil {
+			writeStringHash(h, string(*tsc.NodeAffinityPolicy))
+		}
+		if tsc.NodeTaintsPolicy != nil {
+			writeStringHash(h, string(*tsc.NodeTaintsPolicy))
+		}
```

Comment:
That is a good catch. I am guessing the hash we were previously computing were not encapsulating all the information we could add?